Periodic_4_hyperbolic_triangulation_2/include/CGAL/Periodic_4_hyperbolic_Delaunay_triangulation_traits_2.h

```diff
@@ -0,0 +1,688 @@
+      std::pair<Hyperbolic_point_2, Hyperbolic_point_2> res2 = Intersection()(c , ell);
+
+      Hyperbolic_point_2 a = res1.first;
+      Hyperbolic_point_2 b = res1.second;
+
+      Hyperbolic_point_2 p = res2.first;
+      Hyperbolic_point_2 q = res2.second;
+
+      double aq = CGAL::sqrt(CGAL::to_double(CGAL::squared_distance(a, q)));
+      double pb = CGAL::sqrt(CGAL::to_double(CGAL::squared_distance(p, b)));
+      double ap = CGAL::sqrt(CGAL::to_double(CGAL::squared_distance(a, p)));
+      double qb = CGAL::sqrt(CGAL::to_double(CGAL::squared_distance(q, b)));
+
+      double hyperdist = std::fabs(std::log(CGAL::to_double((aq*pb)/(ap*qb))));
+
+      return hyperdist;
+    }
+  };
+
+  Compute_approximate_hyperbolic_diameter
+  compute_approximate_hyperbolic_diameter_object() const
+  { return Compute_approximate_hyperbolic_diameter(); }
+
+  Construct_hyperbolic_point_2
+  construct_hyperbolic_point_2_object() const
+  { return Construct_hyperbolic_point_2(); }
+
+  Construct_hyperbolic_segment_2
+  construct_hyperbolic_segment_2_object() const
+  { return Construct_hyperbolic_segment_2(); }
+
+  // wrong names kept for demo
+  Construct_segment_2
+  construct_segment_2_object() const
+  { return Construct_segment_2(); }
+
+  class Construct_hyperbolic_line_2
+  {
+    typedef typename Kernel::Construct_weighted_circumcenter_2   Construct_weighted_circumcenter_2;
+    typedef typename Kernel::Weighted_point_2                    Weighted_point_2;
+    typedef typename Kernel::Point_2                             Bare_point;
+    typedef typename Kernel::Circle_2                            Circle_2;
+
+  public:
+    Construct_hyperbolic_line_2() { }
+
+    Hyperbolic_segment_2 operator()(const Hyperbolic_point_2& p, const Hyperbolic_point_2& q)
+    {
+      Origin o;
+      if(Euclidean_collinear_2()(p, q, Hyperbolic_point_2(o)))
+      {
+        std::pair<Hyperbolic_point_2,Hyperbolic_point_2> inters = Construct_intersection_2()(Euclidean_line_2(p,q), Circle_2(Hyperbolic_point_2(0,0),1));
+        return Euclidean_segment_2(inters.first, inters.second);
+      }
+
+      Weighted_point_2 wp(p);
+      Weighted_point_2 wq(q);
+      Weighted_point_2 wo(Hyperbolic_point_2(o), FT(1)); // Poincaré circle
+
+      Bare_point center = Construct_weighted_circumcenter_2()(wp, wo, wq);
+      FT sq_radius = Compute_squared_Euclidean_distance_2()(p, center);
+
+      Circle_2 circle(center, sq_radius);
+      std::pair<Hyperbolic_point_2,Hyperbolic_point_2> inters = Construct_intersection_2()(circle,Circle_2(Hyperbolic_point_2(0,0),1));
+
+      if(Orientation_2()(circle.center(), inters.first, inters.second) == POSITIVE) {
+        return Circular_arc_2(circle, inters.first, inters.second);
+      } else {
+        return Circular_arc_2(circle, inters.second, inters.first);
+      }
+    }
+  };
+
+  Construct_hyperbolic_line_2
+  construct_hyperbolic_line_2_object() const
+  { return Construct_hyperbolic_line_2(); }
```

Comment:
For all these predicates / constructions, you either don't need them, or should do the same as in Hyperbolic_Delaunay_triangulation_traits.

Reply:
Done, I have changed the traits class to follow Hyperbolic_Delaunay_triangulation_traits.